File: scripts/test_insert.ts

import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

dotenv.config({ path: path.join(__dirname, '../.env') });

const supabaseUrl = process.env.VITE_SUPABASE_URL;
const supabaseKey = process.env.VITE_SUPABASE_ANON_KEY;

console.log("URL:", supabaseUrl ? supabaseUrl.substring(0, 15) + "..." : "MISSING");

const supabase = createClient(supabaseUrl, supabaseKey);

async function testInsert() {
    console.log("Testing Insert...");
    const slug = `test-${Date.now()}`;
    const { data, error } = await supabase.from('locations').insert({
        name: 'Test State',
        slug: slug,
        type: 'state',
        state_code: 'TS',
        hierarchy_path: [slug]
    }).select().single();

    if (error) {
        console.error("Insert Error:", error);
    } else {
        console.log("Insert Success:", data);
        // Clean up
        await supabase.from('locations').delete().eq('id', data.id);
    }
}

testInsert();

File: scripts/test_insert.py
import os
import uuid
import re
from datetime import datetime
from supabase import create_client
from dotenv import load_dotenv

load_dotenv(r'c:\Users\Nick\Downloads\hitmaker-2026\emergency-tradesmen\.env')

url = os.getenv("VITE_SUPABASE_URL")
key = os.getenv("SUPABASE_SERVICE_ROLE_KEY") or os.getenv("VITE_SUPABASE_ANON_KEY")
sb = create_client(url, key)

biz = {
    "name": "Test Entry " + str(uuid.uuid4())[:4],
    "phone": "9999999999",
    "website": "http://example.com"
}
trade = "plumber"
city = "TestCity"

slug = f"{re.sub(r'[^a-z0-9]+', '-', biz['name'].lower()).strip('-')}-{str(uuid.uuid4())[:8]}"
data = {
    "id": str(uuid.uuid4()), 
    "name": biz['name'], 
    "slug": slug,
    "trade": trade, 
    "city": city, 
    "country_code": "US",
    "phone": biz['phone'], 
    "address": "123 Test St",
    "verified": True, 
    "tier": "standard", 
    "created_at": datetime.now().isoformat(),
    "website": biz.get('website'),
    "social_links": {} # This is the suspected problem column
}

print(f"Attempting insert with social_links...")
try:
    res = sb.table("businesses").insert(data).execute()
    print("Insert SUCCESS")
except Exception as e:
    print(f"Insert FAILED: {e}")

print(f"\nAttempting insert WITHOUT social_links...")
data.pop("social_links")
data["id"] = str(uuid.uuid4())
data["slug"] = data["slug"] + "-no-social"
try:
    res = sb.table("businesses").insert(data).execute()
    print("Insert WITHOUT social_links SUCCESS")
except Exception as e:
    print(f"Insert WITHOUT social_links FAILED: {e}")
